src/Execute/ExecutionEnv.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Command} from '../Backend/Command';
+import {Executor} from '../Backend/Executor';
+import {gToolchainEnvMap} from '../Toolchain/ToolchainEnv';
+
+// Specify Device with 'name' on `getEnableEnvList()` and `ExecutionEnv`.
+interface ExecutionEnv {
+  name(): string;
+  host(): string;
+  listExecutableExt(): string[];
+  isExecutionEnvAvail(name: string): boolean;
+  getEnableEnvList(): string[];
+  getInferenceCmd(model_path: string, _options?: string[]): Command;
+}
+
+class ToolchainExecutorEnv implements ExecutionEnv {
+  envName: string;  // Specified name for Each Env
+  simulator: Executor;
```

Comment:
(Q) Will this property be valid only if it's a simulator?

Reply:
If it's only about a simulator, how about 'SimulatorEnv' or something instead of 'ToolchainExecutorEnv'?

Reply:
no, I just name this as a `simulator`, but this property comes from `toolchain` executor. if any name for this, please suggest!

Reply:
> If it's only about a simulator,

um... I just name this as simulator, but I thought this comes from all executor comes from `toolchain` that is installed on.

> I think that TizenEnv kinda executes the files from compiled the toolchain, 

I think this is right, as I know.

>so ToolchainExecutorEnv sounds to be a superset of TizenEnv.

um.... I thought `ToolchainExecutorEnv` based on toolchain, and `TizenEnv` based on tizen.
so, follow from this, what about `TizenExectuionEnv`?

Reply:
> um... I just name this as simulator, but I thought this comes from all executor comes from toolchain that is installed on.

Toolchain could contain many things (simulator, emulator, ... runtume, ...) depending on backend vendors.
Another approach could be renaming this to `SimulatorEnv`. 😸 
So, I guess specific name like `SimulatorEnv` might be better.... we might ask backend vendors to implement the details of its method.

Reply:
While looking at `TizenEnv`... one question came up to my mind. 

If a company named FOO wants to create a backend extension for their special emulator `FooEmulator` which I guess implements `interface Executor`, 
should we have `FooExecutionEnv` in ONE-vscode extension to support `FooEmulator`?

Reply:
umm.... I think that `tizenEnv` makes some confusing point. I thought this could be a base class like `compilerBase` or 'executorEnv' to use in backend. I think I need to change this as a `ExecutionEnvBase`.

> should we have `FooExecutionEnv` in ONE-vscode extension to support `FooEmulator`?

then if someone wants to add a new emulator on this, just create new class extends `ExecutionEnvBase` on their backend.